Libraries/NativeAnimation/RCTNativeAnimatedNodesManager.m

```diff
@@ -28,6 +28,21 @@
 #import "RCTValueAnimatedNode.h"
 #import "RCTTrackingAnimatedNode.h"
 
+// We do some normalizing of the event names in RCTEventDispatcher#RCTNormalizeInputEventName.
+// To make things simpler just get rid of the parts we change in the event names we use here.
+// This is a lot easier than trying to denormalize because there would be multiple possible
+// denormalized forms for a single input.
+NSString *RCTNormalizeAnimatedEventName(NSString *eventName)
```

Comment:
We already have same function defined in `Base/RCTEventDispatcher.m$21`.
Are they special reasons why we should duplicate this code here?

Reply:
They are actually slightly different, this one just removes the prefix and the one in `Base/RCTEventDispatcher.m$21` replaces `on` with `top`. Basically when happens is that when events are registered / unregistered it is done with the `on` prefix and when they are dispatched it is with the `top` prefix. This just ignore prefixes when creating they key for the event mapping.

The only case this could be problematic is if a component has a `onFoo` and a `topFoo` prop but I think it is fine. The price to pay for the mess we currently have with the event name prefixes...